### PR TITLE
Add `gp timeout show` command

### PR DESCRIPTION
**Pull request by andrew-farries**
## Description
Add a `gp timeout show` command to the `gp` CLI to go along with the recently added `gp timeout extend` command (see https://github.com/gitpod-io/gitpod/pull/10619).

- [x] ~~Need to hold until PR https://github.com/gitpod-io/gitpod/pull/10896 deployed~~
- [x] Maybe want to follow up @filiptronicek 's suggestion https://github.com/gitpod-io/gitpod/pull/10782#issuecomment-1162703143 

## How to test

1. Create a workspace in preview environment
2. Run `gp timeout show`
3. Output should be (value depends on user's plan):
  ```
  60m
  ```
4. Extend the workspace timeout with `gp timeout extend`. 
5. Rerun `gp timeout show`. Output should be:
  ```
  180m
  ```
6. Open another workspace, and extend its timeout `gp timeout extend`
7. Back to first one, `gp timeout show` should back to original one

## Release Notes

```release-note
Add command `gp timeout show` to show the timeout of current workspace
```

## Documentation

We'll need to add something to https://github.com/gitpod-io/website/pull/2218 either before or after it's merged.

## Werft options:

- [x] /werft with-preview

### CODE REVIEW

File: components/gitpod-cli/cmd/timeout-show.go
Comment (akosyakov) [inline components/gitpod-cli/cmd/timeout-show.go:1]: 2022

**Comment by akosyakov** [comment]
@andrew-farries Don't forget about docs update. At least file an issue.

**Comment by andrew-farries** [comment]
@mustard-mh Do you want to add the docs for this to your existing docs PR (https://github.com/gitpod-io/website/pull/2218) or would you rather I create a new one?

**Comment by mustard-mh** [comment]
> @mustard-mh Should we add the docs for this to your existing docs PR (https://github.com/gitpod-io/website/pull/2218) or would you rather I create a new one?

@andrew-farries  You can do it after previous one https://github.com/gitpod-io/website/pull/2218 merged

**Review by mustard-mh — COMMENTED** [review]
I tested with it. As we can see in image below, 

default value is `long`
after extend `180m`
extend another workspace, this workspace will reset to default `long`

If we can't format every output time to `180m`, maybe we can turn `180m` to `extended`? 

PS. As a user I prefer `180m` formated duration🙂

https://github.com/gitpod-io/gitpod/blob/7bf63df2e9f39bf80fd83e1ae0d1a864a2e69c9a/components/gitpod-protocol/src/gitpod-service.ts#L340-L349

image

**Comment by mustard-mh** [comment]
https://github.com/gitpod-io/gitpod/pull/10782#pullrequestreview-1013290002 What do you think/prefer? @loujaybee

**Review by mustard-mh — COMMENTED** [review]
If we want to turn all output into `180m` formatted, maybe we need a map/func in protocol, since we have multiple similar code like this below
https://github.com/gitpod-io/gitpod/blob/402894b63b604234899db345fca0a9ce3750cec1/components/server/src/user/user-service.ts#L173-L183
https://github.com/gitpod-io/gitpod/blob/402894b63b604234899db345fca0a9ce3750cec1/components/server/src/user/user-service.ts#L185-L196

**Comment by mustard-mh** [comment]
Notice that if we need changes from server, this PR will be approved before next WebApp deploy, or it will block IDE deployment 🙏

**Comment by mustard-mh** [comment]
> Thanks a lot @andrew-farries for the new command 🧡! I tested it and ti works as advertised.
> 
> Maybe we could add to the minutes just a "status" which would indicate whether the timeout is the default one or the extended one? Something like
> 
> ```
> $ gp timeout show
> 60m (Default)
> $ gp timeout extend
> $ gp timeout show
> 180m (Extended)
> ```

@filiptronicek  Remember that `60m` is not default for all users, value of default is depended on Plan of user/team. cc @andrew-farries 

|Image|
|-|
|image|

**Comment by andrew-farries** [comment]
> Remember that 60m is not default for all users, value of default is depend on Plan of user/team.

I suggest we leave the output as is for now, ie without `(Default)` or `(Extended)`.

**Comment by filiptronicek** [comment]
> Remember that 60m is not default for all users, value of default is depend on Plan of user/team.

@mustard-mh I agree, thanks for bringing this up. I wanted to say, that as a user, I would personally like to check `gp timeout show` just to see how many minutes my timeout is and whether I have successfully for example extended it for this repo. For checking the subscription, there could be something like `gp billing` (not sure if suggested before).

> I suggest we leave the output as is for now, ie without (Default) or (Extended).

@andrew-farries I see where you're coming from, it could perhaps be in some way programmatically consumed afterwards as well, but I would prefer to see a bit more context about my timeout, if we have it available. We could also ship it as is and afterwards provide a flag or similar to provide such information.

**Comment by mustard-mh** [comment]
If a user cannot extend timeout, output will be like this @filiptronicek 

Only **Unleashed** plan can extend timeout 🙈

|Image|
|-|
|image|

**Comment by mustard-mh** [comment]
Could you rebase and squash first and third commit since they are not worth split. cc @andrew-farries 

image

**Comment by filiptronicek** [comment]
> If a user cannot extend timeout, output will be like this @filiptronicek

@mustard-mh yes, I meant like if you have a lot of workspaces open and want to check a certain one to see if 15 minutes ago you correctly extended its timeout, you can check if it's extended. I just feel like having that bit of extra info would be nice from this command - some new users may not know if 180 minutes is extended or standard.

**Review by mustard-mh — APPROVED** [review]
Tested and code LGTM

/hold

- [ ] ‼️ Need to deploy WebApp first, so this PR should unhold before next WebApp deploy
- [ ] Maybe want to follow up @filiptronicek 's suggestion https://github.com/gitpod-io/gitpod/pull/10782#issuecomment-1162703143 

I'll add them to PR description too

**Comment by mustard-mh** [comment]
> > @mustard-mh Should we add the docs for this to your existing docs PR ([gitpod-io/website#2218](https://github.com/gitpod-io/website/pull/2218)) or would you rather I create a new one?
> 
> @andrew-farries You can do it after previous one [gitpod-io/website#2218](https://github.com/gitpod-io/website/pull/2218) merged, I can ping you after that

@andrew-farries  Previous doc PR has been merged, you can start it or ask @filiptronicek to help with it 👍

**Comment by andrew-farries** [comment]
Docs PR is here: https://github.com/gitpod-io/website/pull/2292

**Comment by mustard-mh** [comment]
Thank you @andrew-farries , could you split this PR into two PRs, to unblock deployment? `server` will approve first, and then `gp-cli`. 

See more context in [internal chat](https://gitpod.slack.com/archives/C01KGM9BH54/p1656053577087659?thread_ts=1656048950.670019&cid=C01KGM9BH54)

**Comment by andrew-farries** [comment]
>  could you split this PR into two PRs, to unblock deployment? `server` will approve first, and then `gp-cli`.

I've done this. This PR now contains just the `gp` change and https://github.com/gitpod-io/gitpod/pull/10896 has the `server` change.

**Review by geropl — APPROVED** [review]
Thx, works as expected! :+1: 

image
nit: I think the `long` in the beginning might be a hint towards a conversion bug in `server`. But as it's a display issue only: :+1:

**Comment by jeanp413** [comment]
Some feedback, not really important, but I would expect some more user friendly output other than just a number, same with `extend` a message telling me what was the side effect of running this command

**Review by mustard-mh — APPROVED** [review]
Thank you, LGTM

**Comment by mustard-mh** [comment]
/unhold

> Some feedback, not really important, but I would expect some more user friendly output other than just a number, same with extend a message telling me what was the side effect of running this command

> I see where you're coming from, it could perhaps be in some way programmatically consumed afterwards as well, but I would prefer to see a bit more context about my timeout, if we have it available. We could also ship it as is and afterwards provide a flag or similar to provide such information.

You can do it in another PR if you want, to print more friendly 🙂 cc @jeanp413 @filiptronicek 

i.e. `extend` success don't print anything🙈, we can show current ws timeout

**Pull request merged**
